File: post/jetFlame/shaddix/old_code_post_tools/tools/compute_pdf.m

function [xbins P] = compute_pdf(x, nbins, xmin, xmax)

    if(nargin==1)
        nbins = 100;
        xmin = min(x);
        xmax = max(x);
    end
    if(nargin==2)
        xmin = min(x);
        xmax = max(x);
    end
    if(nargin==3)
        error('wrong number of arguments');
    end

    dxbin = (xmax-xmin)/nbins;
    xbins = linspace(xmin+dxbin/2, xmax-dxbin/2, nbins)';

    ibin = floor((x-xmin)/dxbin) + 1;
    ibin(find(ibin<1)) = 1;
    ibin(find(ibin>nbins)) = nbins;

    for i=1:nbins
        P(i) = length(find(ibin==i));
    end

    P = P/length(x)/dxbin;

end

File: post/jetFlame/shaddix/old_code_post_tools/tools/compute_pdf.py
import numpy as np

def compute_pdf(x, xmin, xmax, nbins=100) :

    dxbin = (xmax-xmin)/nbins;
    xbins = np.linspace(xmin+dxbin/2, xmax-dxbin/2, num=nbins)

    ibin = np.floor((x-xmin)/dxbin)
    np.clip(ibin,0,nbins-1)

    P = np.zeros(nbins)

    for i in range(nbins) :
        P[i] = np.size(np.where(ibin==i))

    P = P/len(x)/dxbin;

    return(P,xbins)

